dist/tools/compile_test/compile_test.py

```diff
@@ -110,6 +110,7 @@ def build_all():
                 results_with_output = list(filter(lambda res: res[2].getvalue(), results_with_output))
                 failed_with_output = list(filter(lambda res: 'failed' in res[0], results_with_output))
                 success_with_output = list(filter(lambda res: 'success' in res[0], results_with_output))
+                check_call(["rm", "-rf", join(riotbase, folder, application, "bin")])
```

Comment:
I'm not comfortable with running `rm -rf` on a directory named `bin`. Can we make this safer somehow?
Can riotbase, folder and application ever end up being empty or "/"?

Reply:
Oh right... >.< I'll try to find something safer

Reply:
(I'm not trusting `make clean`)

Reply:
Done